include/rabbitmq-c/amqp.h

```diff
@@ -2433,6 +2433,26 @@ AMQP_EXPORT
 int AMQP_CALL amqp_set_rpc_timeout(amqp_connection_state_t state,
                                    const struct timeval *timeout);
 
+
+/**
+ * amqp_publisher_confirm_wait
+ *
+ * Wait for a publisher confirm when the connection is in select mode.
+ * If the response has a `reply_type` of `AMQP_RESPONSE_LIBRARY_EXCEPTION` _and_
+ * the `library_error` is `AMQP_STATUS_UNEXPECTED_STATE`, then the frame
+ * received was likely not an ack.
+ *
+ * \param [in] state connection state
+ * \param [out] pointer to where the ACK details should go
+ * \param [in] timeout when waiting for the frame. Passing NULL will result in
+ * blocking behavior
+ * \returns amqp_rpc_reply_t *
+ */
+AMQP_EXPORT
+amqp_rpc_reply_t AMQP_CALL amqp_publisher_confirm_wait(
+    amqp_connection_state_t state, amqp_envelope_t *envelope,
+    amqp_basic_ack_t *ack, const struct timeval *timeout);
```

Comment:
"out" parameters should go after any "in" parameters (so this should go after timeout).

Alternatively consider making this function return `amqp_basic_ack_t*`, returning NULL on error, and making the caller consult amqp_get_rpc_reply(...) when NULL is returned. (I haven't thought through the ergonomics of an API like this, what you have may be preferable).

Reply:
The other RPC-like methods use `amqp_simple_rpc_decoded`, which you can use for inspiration if this ends up being the preferred way of implementing this: https://github.com/alanxz/rabbitmq-c/blob/a2d0be96d6902fc909f2cc9a80fb385b4f9362c9/librabbitmq/amqp_socket.c#L1110 to retur

Reply:
I'll take a whack at this, thanks.

Reply:
Yeah, I still kinda like my approach better for a few reasons:
- The ACK belongs to the library, so the application can't use the `amqp_basic_ack_t*` that is decoded. By having the application allocate their own and then put the data there lets the application own the memory there which seems like it'll make using this in threading more straight-forward.
- For some reason, I see this as more analogous to `amqp_consume_message()` than `amqp_simple_rpc_decoded()`. This could be some misunderstanding on my part, and I'd welcome a different perspective.

I've pushed changes for the parameter ordering piece, let me know if you have any additional tweaks you'd like to see.

Reply:
Hey @alanxz, let me know if there's anything more you'd like me to change here.

Reply:
Hey @alanxz, let me know if there's anything more you'd like me to change here.